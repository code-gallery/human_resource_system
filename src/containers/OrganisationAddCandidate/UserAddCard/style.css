.UserAddCard {
  box-sizing: border-box;
  position: relative;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin: 0;
  padding: 11px 30px 11px 19px;
  font-family: "Graphik-Regular", sans-serif;
  font-weight: 400;
  -webkit-font-smoothing: antialiased;
  border: 1px solid #E7E5EE;
  border-top: none;
  background-color: #fff;
}

.UserAddCard__text {
  overflow: hidden;
  margin-right: 40px;
}

.UserAddCard.is-top {
  border-top: 1px solid #E7E5EE;
}

.UserAddCard__name {
  margin-bottom: 2px;
  font-size: 15px;
  line-height: 18px;
  color: #212224;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.UserAddCard__location {
  font-size: 13px;
  line-height: 16px;
  color: #7F8DAA;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.UserAddCard__btn {
  all: initial;
  box-sizing: border-box;
  display: block;
  margin-left: auto;
  font-family: "Graphik-Medium";
  font-weight: 600;
  -webkit-font-smoothing: antialiased;
  font-size: 11px;
  line-height: 13px;
  text-transform: uppercase;
  letter-spacing: 0.075em;
  color: #498DF0;
  cursor: pointer;
}

.UserAddCard__btn.is-disabled {
  color: #E7E5EE;
  cursor: not-allowed;
}

.UserAddCard__avatar {
  display: block;
  margin-right: 17px;
  width: 50px;
  min-width: 50px;
  height: 50px;
  overflow: hidden;
}
