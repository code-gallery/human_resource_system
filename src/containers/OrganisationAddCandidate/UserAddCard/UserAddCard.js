import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import isString from 'lodash/isString'
import Avatar from 'components/$Avatar'
import './style.css'

const UserAddCard = ({
  top,
  profileImage,
  name,
  tagline,
  added,
  onAdd,
  className
}) => {
  const style = classNames(
    'UserAddCard',
    { 'is-top': top },
    className
  )

  return (
    <section className={style}>
      <div className="UserAddCard__avatar">
        <Avatar imgUrl={profileImage} alt={name} />
      </div>

      <div className="UserAddCard__text">
        <div className="UserAddCard__name">{name}</div>
        <div className="UserAddCard__location">
          {isString(tagline) ? tagline : ''}
        </div>
      </div>

      {
        added
          ? <button disabled className="UserAddCard__btn is-disabled">ADDED</button>
          : <button className="UserAddCard__btn" onClick={onAdd}>ADD</button>
      }
    </section>
  )
}

UserAddCard.propTypes = {
  profileImage: PropTypes.string,
  name: PropTypes.string.isRequired,
  tagline: PropTypes.string,
  added: PropTypes.bool.isRequired,
  onAdd: PropTypes.func.isRequired,
  top: PropTypes.bool,
  className: PropTypes.string
}

UserAddCard.defaultProps = {
  top: false,
  className: ''
}

export default UserAddCard
