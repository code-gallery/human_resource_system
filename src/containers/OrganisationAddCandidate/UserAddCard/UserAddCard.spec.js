import React from 'react'
import renderer from 'react-test-renderer'
import UserAddCard from './UserAddCard'

describe('<UserAddCard />', () => {
  const props = {
    name: 'James Carter',
    added: true,
    onAdd: jest.fn()
  }

  describe('Rendering', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <UserAddCard {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
