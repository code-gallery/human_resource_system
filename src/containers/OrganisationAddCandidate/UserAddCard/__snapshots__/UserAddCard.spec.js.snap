// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserAddCard /> Rendering renders correctly 1`] = `
<section
  className="UserAddCard"
>
  <div
    className="UserAddCard__avatar"
  >
    <img
      alt="James Carter"
      className="$Avatar"
      src="https://collaborativecbt.com/wp-content/uploads/2016/12/default-avatar-768x768.png"
    />
  </div>
  <div
    className="UserAddCard__text"
  >
    <div
      className="UserAddCard__name"
    >
      James Carter
    </div>
    <div
      className="UserAddCard__location"
    >
      
    </div>
  </div>
  <button
    className="UserAddCard__btn is-disabled"
    disabled={true}
  >
    ADDED
  </button>
</section>
`;
